src/ParseObject.js

```diff
@@ -2228,7 +2231,9 @@ const DefaultController = {
                 return params;
               })
             }, options);
-          }).then((response, status) => {
+          }).then((response) => {
+            const status = response.status;
+            delete response.status;
             batchReturned.resolve(response, status);
```

Comment:
Great catch! It does look weird that a promise will resolve to more than one parameter.

`batchReturned.resolve(response, status);` shouldn't have 2 parameters. Can you write a test case to ensure that the status works.